### PR TITLE
Plugin configurations

### LINKED ISSUE #105: Plugin configuration

### PR DESCRIPTION
**Pull request by olijeffers0n**
Work on a guide for plugin configurations. I made these on a plane so some sections may be lacking. Let me know of any additions needed on the PR or discord :)

closes #105

### CODE REVIEW

**Comment by olijeffers0n** [comment]
Good points, should now be addressed.

**Review by Owen1212055 — APPROVED** [review]

File: docs/paper/dev/api/plugin-configs.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+        data.put("chunk-x", this.chunkX);
+        data.put("chunk-y", this.chunkZ);
+        data.put("name", this.name);
+        
+        return data;
+    }
+    
+    public static TeleportOptions deserialize(Map<String, Object> args) {
+        return new TeleportOptions((int) args.get("chunk-x"), (int) args.get("chunk-x"), (String) args.get("name"));
```
Comment (Lulu13022002) [inline docs/paper/dev/api/plugin-configs.md:110]: This deserialize two times the chunk X?

File: docs/paper/dev/api/plugin-configs.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    private int chunkX;
+    private int chunkZ;
+    private String name;
+    
+    public TeleportOptions(int chunkX, int chunkZ, String name) {
+        // Set the values
+    }
+    
+    public Map<String, Object> serialize() {
+        Map<String, Object> data = new HashMap();
```
Comment (Lulu13022002) [inline docs/paper/dev/api/plugin-configs.md:100]: Generally you put bracket when using generic with known type (new HashMap<>())

File: docs/paper/dev/api/plugin-configs.md (hunk 1 above)
Comment (Lulu13022002) [reply docs/paper/dev/api/plugin-configs.md]: Reseeing that why the name is chunk-y for a chunk Z?

Comment (olijeffers0n) [reply docs/paper/dev/api/plugin-configs.md]: Good Point, fixed

File: docs/paper/dev/api/plugin-configs.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
+---
+slug: /dev/plugin-configurations
+---
+
+# Plugin Configurations
+
+Configuration files allow users to change certain behaviour and functionality of Plugins. This guide will outline how to use them.
```
Comment (MiniDigger) [inline docs/paper/dev/api/plugin-configs.md:7]: > behaviour 
do we use english or british on docs?

File: docs/paper/dev/api/plugin-configs.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+YAML works by having a tree-like `key: value` pair structure as you would have seen in your [plugin.yml](../getting-started/plugin-yml.md). An example would look like this:
+
+```yaml
+root:
+  one-key: 10
+  another-key: David
+```
+
+When accessing indented values, you separate the levels with "."'s. For example, the key for the "David" string would be `root.another-key`
```
Comment (MiniDigger) [inline docs/paper/dev/api/plugin-configs.md:21]:
```suggestion
When accessing indented values, you separate the levels with `.`. For example, the key for the "David" string would be `root.another-key`
```
looks nicer I think

**Comment by olijeffers0n** [comment]
CI is bugged. This PR is ready for final review and does not actually fail deployment locally

**Comment by MiniDigger** [comment]
fixed

**Review by NotMyFault — APPROVED** [review]

**Pull request merged**